forgerock-openbanking-common/src/main/java/com/forgerock/openbanking/common/CookieHttpSecurityConfiguration.java

```diff
@@ -0,0 +1,79 @@
+ * under the License.
+ */
+package com.forgerock.openbanking.common;
+
+import com.forgerock.openbanking.jwt.services.CryptoApiClient;
+import dev.openbanking4.spring.security.multiauth.configurers.MultiAuthenticationCollectorConfigurer;
+import dev.openbanking4.spring.security.multiauth.configurers.collectors.PSD2Collector;
+import dev.openbanking4.spring.security.multiauth.configurers.collectors.StaticUserCollector;
+import dev.openbanking4.spring.security.multiauth.model.CertificateHeaderFormat;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+
+import java.util.Collections;
+
+import static com.forgerock.openbanking.common.CertificateHelper.CLIENT_CERTIFICATE_HEADER_NAME;
+
+/**
+ * A utility class providing helper methods related to certificates.
+ */
+public class CookieHttpSecurityConfiguration {
+
+    /**
+     * Configures the instance of {@link HttpSecurity} with the internal and external OBRI certificates, plus the {@link DecryptingJwtCookieCollector}.
+     *
+     * @param httpSecurity the {@link HttpSecurity} to configure.
+     * @param obriInternalCertificates the {@link OBRIInternalCertificates} to add as a collector.
+     * @param obriExternalCertificates the {@link OBRIExternalCertificates} to add as a collector.
+     * @param cryptoApiClient the {@link CryptoApiClient} for the {@link DecryptingJwtCookieCollector} collector.
+     */
+    public static void configureHttpSecurity(HttpSecurity httpSecurity, OBRICertificates obriInternalCertificates, OBRICertificates obriExternalCertificates, CryptoApiClient cryptoApiClient) throws Exception {
+        JwtCookieAuthorityCollector jwtCookieAuthorityCollector = new JwtCookieAuthorityCollector();
+        httpSecurity
+                .csrf().disable()
+                .authorizeRequests()
+                .anyRequest()
+                .permitAll()//.authenticated()
+                .and()
+                .authenticationProvider(new CustomAuthProvider())
+                .apply(new MultiAuthenticationCollectorConfigurer<HttpSecurity>()
+                        .collector(PSD2Collector.psd2Builder()
+                                .collectFromHeader(CertificateHeaderFormat.JWK)
+                                .headerName(CLIENT_CERTIFICATE_HEADER_NAME)
+                                .usernameCollector(obriInternalCertificates)
+                                .authoritiesCollector(obriInternalCertificates)
+                                .build())
+                        .collector(PSD2Collector.psd2Builder()
+                                .collectFromHeader(CertificateHeaderFormat.JWK)
+                                .headerName(CLIENT_CERTIFICATE_HEADER_NAME)
+                                .usernameCollector(obriExternalCertificates)
+                                .authoritiesCollector(obriExternalCertificates)
+                                .build())
+                        .collector(DecryptingJwtCookieCollector.builder()
```

Comment:
This is Jorge's fix - applied to the common configuration (see the PR's description for a list of affected classes).